src/scripts/cronjobs/emailBreachAlerts.tsx

```diff
@@ -260,7 +264,30 @@ export async function poll(
                 });
 
                 const l10n = getCronjobL10n(sanitizeSubscriberRow(recipient));
-                const subject = l10n.getString("breach-alert-subject");
+                const enabledFeatureFlags = await getEnabledFeatureFlags({
+                  email: recipient.primary_email,
+                });
+                const subject = enabledFeatureFlags.includes(
+                  "BreachEmailRedesign",
+                )
+                  ? l10n.getString("email-breach-alert-all-subject")
+                  : l10n.getString("breach-alert-subject");
+
+                /**
+                 * Without an active user session, we don't know the user's country. This is
+                 * our best guess based on their locale. At the time of writing, it's only
+                 * used to determine whether to count SSN breaches (which we don't have
+                 * recommendations for outside the US).
+                 */
+                const assumedCountryCode = getSignupLocaleCountry(recipient);
+
+                const scanData = await getLatestOnerepScanResults(
+                  recipient.onerep_profile_id,
+                );
+                const allSubscriberBreaches = await getSubscriberBreaches({
+                  fxaUid: recipient.fxa_uid,
+                  countryCode: assumedCountryCode,
+                });
```

Comment:
@rhelmer @mansaj I wanted to draw some extra attention to these - I'm not sure how many subscribers we might be sending emails for in a single poll, but the email can now show the number of exposed data points for the user, which means I have to fetch all scan results and breaches for each of them. Just wondering if this won't cause excessive load and, if so, how best to deal with that?

Reply:
Here's some background info, I'm going to add advice as a separate comment but in case this is useful:

It's a bit hard to predict how many subscribers we'll email for a single `poll()` invocation - there's some analysis of the existing HIBP anonymized dataset here https://blog.cloudflare.com/validating-leaked-passwords-with-k-anonymity/

The `receivedMessage` being processed here is the same hash prefix + suffixes described in that blog post. Our system additionally only sends if certain criteria are met, and also of course each hash must match an email in the Monitor database so in practice we send fewer than the average 478 (max 584) cited there.

However the possible number is quite large:
> it is worth noting that as the length of a SHA-1 hash is a total of 40 hexadecimal characters long and 5 characters is utilised by the Hash Prefix, the total number of possible hashes associated with a Hash Prefix is 16^{35} ≈ 1.39E42.

Reply:
We run the jobs that process these messages in parallel, so it's OK if one takes a while, and it can be safely resumed if interrupted. So I wouldn't worry too much about the edge cases here, although we don't want to have too much of a delay.

If we're using cached data I wouldn't expect it to be a problem, but if we need to refresh it from the provider that might cause more of an issue. Now that it's easy for us to use Redis again I'd suggest we consider caching data, we could have Redis automatically expire data from the cache after a certain amount of time.

I'm not sure we'd get a very good hit rate on that cache though, the main problem I can foresee is if we're refreshing broker data (I'm assuming that's what you mean by "scan results"?)

Reply:
I refactored the code to only fetch the above data for free US users who have run a scan, which I think is (at this point in time) a relatively manageable number. I'm still not sure if it was even going to be a problem, but at least this should make it safer to roll out: a7d419d.